addon/components/ember-table.js

```diff
@@ -0,0 +1,564 @@
+    var contentHeight = this.get('_tableContentHeight') +
+        this.get('_headerHeight') + this.get('_footerHeight');
+    return height < contentHeight;
+  }).property('_height', '_tableContentHeight', '_headerHeight',
+      '_footerHeight'),
+
+  _hasHorizontalScrollbar: Ember.computed(function() {
+    var contentWidth = this.get('_tableColumnsWidth');
+    var tableWidth = this.get('_width') - this.get('_fixedColumnsWidth');
+    return contentWidth > tableWidth;
+  }).property('_tableColumnsWidth', '_width', '_fixedColumnsWidth'),
+
+  // tables-container height adjusts to the content height
+  _tablesContainerHeight: Ember.computed(function() {
+    var height = this.get('_height');
+    var contentHeight = this.get('_tableContentHeight') +
+        this.get('_headerHeight') + this.get('_footerHeight');
+    return Math.min(contentHeight, height);
+  }).property('_height', '_tableContentHeight', '_headerHeight',
+      '_footerHeight'),
+
+  // Actual width of the fixed columns
+  _fixedColumnsWidth: Ember.computed(function() {
+    return this._getTotalWidth(this.get('fixedColumns'));
+  }).property('fixedColumns.@each.width'),
+
+  // Actual width of the (non-fixed) columns
+  _tableColumnsWidth: Ember.computed(function() {
+    // Hack: We add 3px padding to the right of the table content so that we can
+    // reorder into the last column.
+    var contentWidth = this._getTotalWidth(this.get('tableColumns')) + 3;
+    var availableWidth = this.get('_width') - this.get('_fixedColumnsWidth');
+    return Math.max(contentWidth, availableWidth);
+  }).property('tableColumns.@each.width', '_width', '_fixedColumnsWidth'),
+
+  _rowWidth: Ember.computed(function() {
+    var columnsWidth = this.get('_tableColumnsWidth');
+    var nonFixedTableWidth = this.get('_tableContainerWidth') -
+        this.get('_fixedColumnsWidth');
+    return Math.max(columnsWidth, nonFixedTableWidth);
+  }).property('_fixedColumnsWidth', '_tableColumnsWidth',
+      '_tableContainerWidth'),
+
+  // Dynamic header height that adjusts according to the header content height
+  _headerHeight: Ember.computed(function() {
+    var minHeight = this.get('minHeaderHeight');
+    var contentHeaderHeight = this.get('_contentHeaderHeight');
+    return Math.max(contentHeaderHeight, minHeight);
+  }).property('_contentHeaderHeight', 'minHeaderHeight'),
+
+  // Dynamic footer height that adjusts according to the footer content height
+  _footerHeight: Ember.computed(function() {
+    return this.get('hasFooter') ? this.get('footerHeight') : 0;
+  }).property('footerHeight', 'hasFooter'),
+
+  _bodyHeight: Ember.computed(function() {
+    var bodyHeight = this.get('_tablesContainerHeight');
+    if (this.get('hasHeader')) {
+      bodyHeight -= this.get('_headerHeight');
+    }
+    if (this.get('hasFooter')) {
+      bodyHeight -= this.get('footerHeight');
+    }
+    return bodyHeight;
+  }).property('_tablesContainerHeight', '_hasHorizontalScrollbar',
```

Comment:
`_hasHorizontalScrollbar` is not used in the function, not sure if it's needed here